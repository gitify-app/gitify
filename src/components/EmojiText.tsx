import path from 'node:path';
import twemoji from '@discordapp/twemoji';
import { type FC, useEffect, useRef } from 'react';
import { getDirectoryPath } from '../utils/helpers';

export interface IEmojiText {
  text: string;
}

type TwemojiOptions = {
  base: string;
  size: string;
  ext: string;
};

export const EmojiText: FC<IEmojiText> = ({ text }) => {
  const ref = useRef(null);

  useEffect(() => {
    if (ref.current) {
      ref.current.innerHTML = twemoji.parse(text, {
        folder: 'svg',
        ext: '.svg',
        callback: (icon: string, options: TwemojiOptions, _variant: string) => {
          const source = path.resolve(
            getDirectoryPath(),
            '../../node_modules/@discordapp/twemoji/dist',
          );

          return ''.concat(source, '/', options.size, '/', icon, options.ext);
        },
      });
    }
  }, [text]);

  return <span ref={ref} />;
};
