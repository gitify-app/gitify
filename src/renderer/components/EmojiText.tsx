import path from 'node:path';
import twemoji from '@discordapp/twemoji';
import { type FC, useEffect, useRef } from 'react';

export interface IEmojiText {
  text: string;
}

type TwemojiOptions = {
  base: string;
  size: string;
  ext: string;
};

export const EmojiText: FC<IEmojiText> = ({ text }) => {
  const ref = useRef(null);

  useEffect(() => {
    if (ref.current) {
      ref.current.innerHTML = twemoji.parse(text, {
        folder: 'svg',
        ext: '.svg',
        callback: (
          icon: string,
          _options: TwemojiOptions,
          _variant: string,
        ) => {
          return path.join('images', 'twemoji', `${icon}.svg`);
        },
      });
    }
  }, [text]);

  return <span ref={ref} />;
};
