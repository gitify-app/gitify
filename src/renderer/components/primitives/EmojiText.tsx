import { type FC, useEffect, useRef } from 'react';

import { convertTextToEmojiImgHtml } from '../../utils/emojis';

export interface IEmojiText {
  text: string;
}

export const EmojiText: FC<IEmojiText> = ({ text }) => {
  const ref = useRef(null);

  useEffect(() => {
    if (ref.current) {
      ref.current.innerHTML = convertTextToEmojiImgHtml(text);
    }
  }, [text]);

  return <span ref={ref} />;
};
