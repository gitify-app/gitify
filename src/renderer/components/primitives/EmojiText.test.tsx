import { render } from '@testing-library/react';
import { EmojiText, type IEmojiText } from './EmojiText';

describe('renderer/components/primitives/EmojiText.tsx', () => {
  it('should render', () => {
    const props: IEmojiText = {
      text: 'üç∫',
    };
    const tree = render(<EmojiText {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
