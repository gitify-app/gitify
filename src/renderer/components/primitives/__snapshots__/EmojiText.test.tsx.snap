// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer/components/primitives/Emoji.tsx should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span>
        <img
          alt="🍺"
          class="emoji"
          draggable="false"
          src="images/twemoji/1f37a.svg"
        />
      </span>
    </div>
  </body>,
  "container": <div>
    <span>
      <img
        alt="🍺"
        class="emoji"
        draggable="false"
        src="images/twemoji/1f37a.svg"
      />
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
